### ISSUE TITLE
Modal contains a focusable element that has no interactive content 

### ISSUE THREAD

**Issue by GrigorM** [ant-design/ant-design#40380]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/antd-reproduction-template-forked-fn2vbq)

### Steps to reproduce

1. Open any modal dialog, either the basic one or the ones created with Modal.method()
2. Open the browser console and type `document.activeElement`
3. The active element is empty and not interactive
```<div tabindex="0" style="width: 0px; height: 0px;… hidden; outline: none;" aria-hidden="true"></div>```

### What is expected?

From [MDN](https://developer.mozilla.org/en-US/docs/Web/Accessibility/Understanding_WCAG/Keyboard#focusable_elements_should_have_interactive_semantics):
> ## Focusable elements should have interactive semantics  
> If an element can be focused using the keyboard, then it should be interactive; that is, the user should be able to do something to it and produce a change of some kind (for example, activating a link or changing an option).

The modal should automatically focus the close button, or the Ok button, and navigation with Tab or Shift Tab should send focus on interactive elements, not on empty and non-interactive ones

### What is actually happening?

An empty and non-interactive element is being focused by default and/or by navigating with Tab or Shift Tab when modal is open

| Environment | Info |
| --- | --- |
| antd | 5.1.6 |
| React | 18 |
| System | any system |
| Browser | any browser |

---
No idea what role that element is serving there, but it's an accessibility problem, and should be removed.

**Comment by MadCcc** [comment]
This element is used for cycling focus elements inside Modal. And confirm button would be focused in `Modal.confirm`.

**Comment by GrigorM** [comment]
You can cycle through the elements even without it. The problem is that it is an empty. non-interactive element that is receiving focus. It is an accessibility violation (no text, no semantics, no action) therefore it should be removed

**Comment by AnathanPham** [comment]
@MadCcc look at this please. https://github.com/react-component/dialog/pull/360

**Comment by guan404ming** [comment]
I would like to work on this please~~
